core/src/main/java/com/dooapp/fxform/controller/NodeController.java

```diff
@@ -29,7 +30,8 @@
 public abstract class NodeController implements Disposable {
 
     private final ObjectProperty<FXFormNode> node = new SimpleObjectProperty<FXFormNode>();
-    private final ChangeListener<FXFormNode> changeListener;
+    private ChangeListener<FXFormNode> changeListener;

```

Comment:
Is it necessary to keep the reference to the changeListener? As long as the node is present I suppose there are no risk to garbage collect it.

Reply:
Yes this is required. Otherwise the listener will be only reference by a WeakReference in the WeakChangeListener and could be garbage collected too soon.
See [WeakChangeListener](https://docs.oracle.com/javase/8/javafx/api/javafx/beans/value/WeakChangeListener.html) javadoc : 

> Note: You have to keep a reference to the ChangeListener, that was passed in as long as it is in use, otherwise it will be garbage collected to soon.

Reply:
So I do not understand the goal of the WeakChangeListener in this particular situation. It's created, added and removed exactly at the same time than the listener reference. Then managing directly the reference would have the same effect no?

Reply:
- The reference to the ChangeListener is kept so that the ChangeListener does not get garbage collected
- The reference to the WeakChangeListener is kept so that we can remove the listener from the FXFormNode in case this is explicitely required like in dispose() which will called if this NodeController is removed form the form

Reply:
As you pointed out during a discussion about this point, this change is actually not required to solve the garbage collection we are fixing. The ChangeListener is used here "internally" by the NodeController and shouldn't a source of memory leaks. I have removed this change.